controllers/storagesystem_controller.go

```diff
@@ -187,6 +188,36 @@ func (r *StorageSystemReconciler) validateStorageSystemSpec(instance *odfv1alpha
 	return nil
 }
 
+func (r *StorageSystemReconciler) addReferenceToRelatedObjects(instance *odfv1alpha1.StorageSystem, object runtime.Object) {
+
+	objMeta, _ := meta.Accessor(object)
+
+	objectRef := corev1.ObjectReference{
+		APIVersion: object.GetObjectKind().GroupVersionKind().Version,
+		Kind:       object.GetObjectKind().GroupVersionKind().Kind,
+		Name:       objMeta.GetName(),
+		Namespace:  objMeta.GetNamespace(),
+		UID:        objMeta.GetUID(),
+	}
+
+	if instance.Status.RelatedObjects == nil {
+		instance.Status.RelatedObjects = []corev1.ObjectReference{objectRef}
+		return
+	}
+
+	for i := range instance.Status.RelatedObjects {
+		if instance.Status.RelatedObjects[i].Kind == objectRef.Kind &&
+			instance.Status.RelatedObjects[i].Name == objectRef.Name &&
+			instance.Status.RelatedObjects[i].Namespace == objectRef.Namespace {
+
+			instance.Status.RelatedObjects[i] = objectRef
+			return
+		}
+	}
+
+	instance.Status.RelatedObjects = append(instance.Status.RelatedObjects, objectRef)
+}
```

Comment:
Rather than doing all this, is there any reason you're not taking the documented CRD approach? https://github.com/openshift/custom-resource-status/blob/master/objectreferences/README.md

Reply:
Done